conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanGenerator.java

```diff
@@ -557,7 +551,10 @@ private static MethodSpec createStaticFactoryMethod(
                     .addAnnotations(ConjureAnnotations.safety(safetyEvaluator.getUsageTimeSafety(field.conjureDef())))
                     .build()));
             // Follow order on adding methods on builder to comply with staged builders option if set
-            sortedEnrichedFields(fields).map(EnrichedField::poetSpec).forEach(spec -> {
+            Stream<EnrichedField> methodArgs = useStagedBuilders
+                    ? fields.stream().sorted(Comparator.comparing(BeanGenerator::fieldShouldBeInFinalStage))
```

Comment:
Since `fields` is an `ImmutableList`, I think `fields.stream()` would be an ordered stream, the sort would be stable, and we don't need `thenComparing(fields::indexOf)`.